### PR TITLE
fixed setting the value of a non-nullable, non-required field to NULL

### PR DESCRIPTION
**Pull request by mihailignatenko**
Also added check if property actually has `mysql.nullable` option

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by raymondfeng** [comment]
@mihailignatenko Thanks for the patch. Would you please add a test case?

File: lib/mysql.js [hunk 1]
```diff
@@ -272,6 +272,11 @@ MySQL.prototype.toFields = function (model, data) {
       if (undefined === value) {
         return;
       }
+      // skip property if it cannot be NULL, but `toDatabase` function returns NULL
+      if(props[key].mysql && props[key].mysql.nullable &&
```
Comment (raymondfeng) [inline lib/mysql.js]: We should test `|| props[key].mysql.nullable === true` too.

**Comment by raymondfeng** [comment]
@mihailignatenko Please also sign CLA - https://cla.strongloop.com/agreements/strongloop/loopback-connector-mysql

**Comment by raymondfeng** [comment]
@mihailignatenko ping

File: lib/mysql.js (hunk 1 above)
Comment (mihailignatenko) [inline lib/mysql.js]: @raymondfeng do you mean that `nullable` can be set to `true` or `false`, not only `"Y"` or `"N"`

Comment (raymondfeng) [inline lib/mysql.js]: I intend to normalize the `nullable` to be a boolean. That's why I suggest that you cover `true` too.

**Comment by mihailignatenko** [comment]
#92

**Comment by jannyHou** [comment]
@mihailignatenko Are you still working on this pr? Could you rebase the code and add some test cases? thanks.

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by jannyHou** [comment]
Thank you so much for contribution, unfortunately the function fixed in this PR is removed, and the fix seems out of date now. I am closing it, feel free to reopen if there is update, thanks for understanding.

**Pull request closed without merging**